docs/handlers.rst

```diff
@@ -147,6 +147,16 @@ The following 3 core cause-handlers are available::
     def my_handler(spec, **_):
         pass
 
+.. note::
+    Kopf's finalizers will be added to the object when there are delete
+    handlers specified. Finalizers block Kubernetes from fully deleting
+    objects, and Kubernetes will only actually delete objects when all
+    finalizers are removed, i.e. only if the Kopf operator is running to
+    remove them (check: :ref:`finalizers-blocking-deletion` for a work-around).
+    If a delete handler is added but finalizers are not required to block the
+    actual deletion, i.e. the handler is optional, the optional argument
+    `optional=True` can be passed to the delete cause decorator.
```

Comment:
Just a little note: it should be double-backticks here for code quoting

```
 ... ``optional=True`` ...
```

Otherwise, with single-backticks, Sphinx treats this as a reference to a chapter/object/class/method/data/constant of that full name ("optional=True"), and will fail to find it, and will issue a warning, and will format it differently (not as a code snippet).

It is not critical, can be skipped for now, but it is better if done right from the beginning rather than fixed later with the docs cleanups.

Reply:
Fixed.